### PR TITLE
Add blitz auth + next guide

### PR DESCRIPTION
**Pull request by beerose**

### CODE REVIEW

File: app/pages/docs/blitz-auth-with-next.mdx [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+To apply the changes to the database and generate Prisma’s TypeScript
+types, run: `yarn prisma migrate dev`
+
+## Adding auth logic {#add-auth-logic}
+
+Now that the setup part is done, we can proceed to implement a simple auth
+logic using Blitz Auth. This guide will cover sign-up, log-in, and
+log-out.
+
+### Add `pages/api/signup.ts` file {#add-signup-route}
```
Comment (dillonraphael) [inline app/pages/docs/blitz-auth-with-next.mdx:207]: We're placing the `pages` directory inside `src` now

File: app/pages/docs/blitz-auth-with-next.mdx [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+Note: there’s no error handling because I’m trying to keep this guide
+minimal and focus only on how to setup Blitz Auth with Next.js. Before
+using it in your applications, you should extend and modify it
+accordingly.
+
+```ts
+// pages/api/signup.ts
+
+import { SecurePassword } from "@blitzjs/auth"
+import { api } from "../../src/blitz-server"
```
Comment (dillonraphael) [inline app/pages/docs/blitz-auth-with-next.mdx:231]: See above

File: app/pages/docs/blitz-auth-with-next.mdx [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+  body: JSON.stringify({ email, password }),
+})
+```
+
+One thing to notice here is the usage of `getAntiCSRFToken`. When making a
+request from the client to an API route, we need to include `anti-csrf`
+header. You can read more about it
+[here](https://blitzjs.com/docs/session-management#manual-api-requests).
+
+### Add `pages/api/login.ts` {#add-login-route}
```
Comment (dillonraphael) [inline app/pages/docs/blitz-auth-with-next.mdx:282]: Same here

File: app/pages/docs/blitz-auth-with-next.mdx [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+correct, we’ll create a new authenticated session as we did in the signup
+handler.
+
+```ts
+// pages/api/login.ts
+
+import { SecurePassword } from "@blitzjs/auth"
+import { AuthenticationError } from "blitz"
+import db from "../../prisma"
+import { api } from "../../src/blitz-server"
```
Comment (dillonraphael) [inline app/pages/docs/blitz-auth-with-next.mdx:295]: And here

**Review by dillonraphael — CHANGES_REQUESTED** [review]

File: app/pages/docs/blitz-auth-with-next.mdx [hunk 5]
```diff
@@ -0,0 +1,671 @@
...
+```ts
+// prisma/index.ts
+import { PrismaClient } from "@prisma/client"
+
+export * from "@prisma/client"
+const db = new PrismaClient()
+export default db
+```
+
+##### Modify Prisma schema {#modify-prisma-schema}
```
Comment (flybayer) [inline app/pages/docs/blitz-auth-with-next.mdx:173]:
```suggestion
##### Modify Prisma schema
```

**Review by flybayer — APPROVED** [review]

**Pull request merged**